src/GraphQL/Builders/FieldBuilder.cs

```diff
@@ -158,6 +158,24 @@ public virtual FieldBuilder<TSourceType, TReturnType> Validate(Action<object> va
         return this;
     }
 
+    /// <inheritdoc cref="ValidateArguments(Func{Validation.FieldArgumentsValidationContext, ValueTask})"/>
+    public virtual FieldBuilder<TSourceType, TReturnType> ValidateArguments(Action<Validation.FieldArgumentsValidationContext> validation)
+    {
+        FieldType.ValidateArguments = ctx => { validation(ctx); return default; };
+        return this;
+    }
+
+    /// <summary>
+    /// Sets argument validation to the field, replacing any existing validation function.
+    /// Runs after all arguments have been coerced and validated as appropriate.
+    /// Applies only to output fields.
+    /// </summary>
+    public virtual FieldBuilder<TSourceType, TReturnType> ValidateArguments(Func<Validation.FieldArgumentsValidationContext, ValueTask> validation)
```

Comment:
Here sync and async are both supported, a benefit over the `Parser` and `Validator` delegates.

Reply:
However, let's say someone wrote async code, such as to validate an id within a database.  Transient database exceptions would be caught and returned to the caller as a validation error, when in fact it should be a server error.  In this situation the caller should be careful to catch any exceptions other than those intended to be returned to the caller.

Reply:
I'm wondering if maybe we should require people to throw `ValidationError`, so unanticipated server errors, intentional or not, are not passed back to the caller.

Parser and Validator are intended to be a single line of code with very simple logic.  Here it is much more likely that dev coding bugs will be present and find their way back to the caller.

And then there's consistency.  To be consistent with Validator it should capture any error.  To be consistent with the rest of GraphQL.NET it should make unrecognized errors.

Reply:
So... okay as-is?   Should I change it?

Reply:
I vote for exeption handler

Reply:
Done; please review.  Note that currently an unhandled error bubbles to the `DocumentExecuter` which means it will not have a location attached to the error.  Fixing this is a bigger problem than it sounds.  Neither the `ValidationContext` nor the `ValidationOptions` class have a reference to the unhandled exception handler, nor do they have access to either the `ExecutionContext` or `ExecutionOptions` which would have it.  But if this were to be added to the `ValidationContext` as an additional property, the exception returned by it is expected to be an `ExecutionError`.  Whereas only `ValidationError` instances can be added to the `ValidationContext`.  So the exception would have to bubble anyway (although it could contain location information).

I chose to ignore that complexity at this time.  Throwing an `ExecutionError` will append the location information, and throwing a `ValidationError` will not only append location information, but will continue to process the remaining validation rules before terminating the request.

As such, throwing a `ValidationError` or calling `ReportError` is the normal way to report validation errors within the argument validation method, but throwing an `ExecutionError` is allowed (only difference is that remaining validation rules will not execute), and throwing any other exception will trigger the unhandled exception handler (with no location information unfortunately).

Maybe for v9 we can improve this scenario a little further.